### PR TITLE
[QRCodeScanner] Fixed Scan Issues and Supabase Queries

### PR DESCRIPTION
**Pull request by philipye314**
### Description

[//]: # "Required - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."
- Now only run supabase query upon entering QRCodeScanner screen, stored in useState variables.
- Checks now just use the useState variables, much cleaner and less spam quering 
### Screenshots

[//]: # "Required for frontend changes, otherwise optional but strongly recommended. Add screenshots of expected behavior - GIFs if you're feeling fancy!"

## How to review

[//]: # 'Required - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?'
- try adding an invalid case, duplicate case, and finally a working case. After adding it to your cases, make sure the error message is "DUPLICATES NOT ALLOWED!"

## Next steps

[//]: # "Optional - What's NOT in this PR, doesn't work yet, and/or still needs to be done. Note any temporary fixes in this PR that should be cleaned up later."
- Push to hi-fi
### Online sources

[//]: # 'Optional - copy links to any tutorials or documentation that was useful to you when working on this PR'

### Related PRs

[//]: # "Optional - related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

CC: @stephaniewong2

### CODE REVIEW

File: src/app/(BottomTabNavigation)/AllCases/QRCodeScanner/index.tsx [hunk 1]
```diff
@@ -44,35 +56,28 @@ function QRCodeScannerScreen() {
...
-        },
-      });
+    if (!validIds.includes(caseId)) {
+      // TODO: Display error toast message
+      setToast('Not a valid QRCODE!');
+      return;
+    }
+    if (userIds.includes(caseId)) {
+      setToast('DUPLICATES NOT ALLOWED!');
+      return;
```
Comment (stephaniewong2) [inline src/app/(BottomTabNavigation)/AllCases/QRCodeScanner/index.tsx]: We need to check `scanned` here still for valid cases because letting a user scan infinitely in a valid case will lead to the same issue of spamming back and forth between the QR Code Scanner screen and Add Case screen if the user clicks cancel. I think it makes most sense to do a `if-else if` clause here -- if (!validIds.includes(caseId)), else if  (userIds.includes(caseId)), else if (!scanned). Let me know if this makes sense!

**Review by stephaniewong2 — CHANGES_REQUESTED** [review]

File: src/app/(BottomTabNavigation)/AllCases/QRCodeScanner/index.tsx [hunk 2]
```diff
@@ -20,9 +20,21 @@ enum permissions {
...
+      const allCases = await getAllCaseIds();
+      setValidIds(allCases);
+      const userCases = await getCaseIdsFromUserId('NO_ID');
+      setUserIds(userCases);
+    };
+    getAllCasesAndValidCases();
+  }, []);
+
   useEffect(() => {
     navigation.addListener('blur', async () => {
```
Comment (stephaniewong2) [inline src/app/(BottomTabNavigation)/AllCases/QRCodeScanner/index.tsx:39]: In these events, let's also add `setToast('')` so that if a user scans an invalid/duplicate case, sees the error, scans a valid case, and clicks cancel, their QR Code Scanner screen won't display an error. Let me know if that doesn't make sense!
